functions/divide.h

```diff
@@ -0,0 +1,56 @@
+#ifndef FUNCTIONS_DIVIDE_H
+#define FUNCTIONS_DIVIDE_H
+
+// Usage: Divide<F, V>
+// Divide F by V
+// If V = 0, returns 0
+// F, V: FUNCTION, 
+// return value: FUNCTION
+
+class BladeBase;
+
+template<class F, class V>
+class DivideBase {
+public:
+  void run(BladeBase* blade) {
+    f_.run(blade);
+    v_.run(blade);
+  }
+  int getInteger(int led) {
+    if (v_.getInteger(led) == 0) {
```

Comment:
We shouldn't call v_.getInteger(led) twice.
We should call it once and store the result in a local variable.
(Same for the SVF function below.)